File: algorandsamples/v2/myjsdemo/indexer/searchTransactionsNote.js
// SearchTransactionsNote.js
// requires algosdk@1.6.1 or higher 
// verify installed version
// npm list algosdk
const algosdk = require('algosdk');
const indexer_token = "";
const indexer_server = "http://localhost";
const indexer_port = 8981;

// const indexer_server = "https://testnet-algorand.api.purestake.io/idx2/";
// const indexer_port = "";
// const indexer_token = {
//     'X-API-key': 'B3SU4KcVKi94Jap2VXkK83xx38bsv95K5UZm2lab',
// }

// Instantiate the indexer client wrapper
let indexerClient = new algosdk.Indexer(indexer_token, indexer_server, indexer_port);

(async () => {  
    //let names = "Russell Anthony";
    // let names = '{"firstName":"John"';
    const enc = new TextEncoder();
    let note = enc.encode("Hello");
  //  let note = enc.encode(names);
 //   let note = enc.encode('[{"_id":"60c9dc007b9206a8a641f2e8"]}]');   
    let s = Buffer.from(note).toString("base64");
    let transactionInfo = await indexerClient.searchForTransactions()
        .minRound(14838981)
        .notePrefix(s).do();
    // const s = Buffer.from(note).toString("base64");
    // let transactionInfo = await indexerClient.searchForTransactions()
    //     .minRound(14037730)
    //     .notePrefix(s).do();
    console.log("Information for Transaction search: " + JSON.stringify(transactionInfo, undefined, 2));
    // create a buffer
    if (transactionInfo.transactions.length > 0)
    {
        console.log("First Match:"); 
        const buff = Buffer.from(transactionInfo.transactions[0].note, 'base64');
        // decode buffer as UTF-8
        const str = buff.toString('utf-8');
        // print normal string
        console.log(str);        
    }


})().catch(e => {
    console.log(e);
    console.trace();
});

File: algorandsamples/v2/mypythondemo/indexer/search_transactions_note.py
# search_transactions_note.py
import base64
import json
# requires Python SDK version 1.3 or higher
from algosdk.v2client import indexer

# instantiate indexer client
# headers = {
#    "Authorization": "Bearer KegWFLYQnBNVeP4oHCX64dObBk8VemzYdNqsnAOIxYQ8aqJLQTYeVDQyZNnx1PZA"
# }
# myindexer = indexer.IndexerClient(indexer_token="", indexer_address="https://indexer-testnet-green.aws.algodev.network", headers=headers)

# algod_address = "https://testnet-algorand.api.purestake.io/idx2"
# algod_token = ""
# headers = {
#    "X-API-Key": "WpYvadV1w53mSODr6Xrq77tw0ODcgHAx9iJBn5tb",
# }
# myindexer = indexer.IndexerClient(
#     algod_token, algod_address, headers)
# myindexer = indexer.IndexerClient(indexer_token="", indexer_address="http://localhost:8981")
# myindexer = indexer.IndexerClient(indexer_token="YddOUGbAjHLr1uPZtZwHOvMDmXvR1Zvw1f3Roj2PT1ufenXbNyIxIz0IeznrLbDsF", indexer_address="https://indexer-testnet.internal.aws.algodev.network")

myindexer = indexer.IndexerClient(indexer_token="", indexer_address="https://testnet.algoexplorerapi.io/idx2/")

import base64

# note_prefix = '{"firstName":"JohnChr"'.encode()
note_prefix = 'JohnChr'.encode()
print(base64.b64encode('JohnChr'.encode()))
# print(base64.b64encode('{"firstName":"JohnChr"'.encode()))
# note_prefix = 'Hello Rus'.encode()
# 14,556,497
# response = myindexer.search_transactions(
#     note_prefix=note_prefix, min_round=14556000, address="GD64YIY3TWGDMCNPP553DZPPR6LDUSFQOIJVFDPPXWEG3FVOJCCDBBHU5A")
response = myindexer.search_transactions(
    note_prefix=note_prefix,min_round=14556000)


print("note_prefix = " +
      json.dumps(response, indent=2, sort_keys=True))


# print first note that matches
if (len(response["transactions"]) > 0):
    print("Decoded note: {}".format(base64.b64decode(
        response["transactions"][0]["note"]).decode()))
    # person_dict = json.loads(base64.b64decode(
    #     response["transactions"][0]["note"]).decode())
    # print("First Name = {}".format(person_dict['firstName']))
